api/foundation/foundation_api.adoc

```diff
@@ -0,0 +1,141 @@
+`org.jboss.tools.foundation.core` and `org.jboss.tools.foundation.ui` plugins. They help plugin developers quickly gain access to:
+
+- Status object creation
+- Logging of messages
+- Tracing
+- Shared Images
+
+A great example of plugins that already use these classes can be found at link:https://github.com/jbosstools/jbosstools-base/blob/master/stacks/plugins/org.jboss.tools.stacks.core/src/org/jboss/tools/stacks/core/StacksCoreActivator.java[StacksCoreActivator] for the Core case, 
+and link:https://github.com/jbosstools/jbosstools-base/blob/master/runtime/plugins/org.jboss.tools.runtime.ui/src/org/jboss/tools/runtime/ui/RuntimeSharedImages.java[RuntimeSharedImages] for an example of our Shared Images functionality. 
+
+=== The Activators
+
+There are two notable superclasses for a new plugin's activator class. They are `org.jboss.tools.foundation.core.plugin.BaseCorePlugin` and `org.jboss.tools.foundation.ui.plugin.BaseUIPlugin`. 
+A new plugin should extend one of these two classes in almost all cases unless prevented by other architectural concerns. 
+
+`BaseCorePlugin` and `BaseUIActivator` grant access to several very useful APIs. See the three following protected methods:
+
+```
+protected IPluginLog pluginLogInternal()        // To easily log messages
+protected StatusFactory statusFactoryInternal() // To create status objects
+protected void registerDebugOptionsListener(String pluginId, AbstractTrace trace, BundleContext bc) // To aid in tracing
+```
+
+Since these three methods are all protected, they are not callable from outside the class heirarchy. 
+A new plugin's activator should expose the plugin log and the status factory methods in whatever way
+they decide is best. Most often, with the goal of concise code for callers, a plugin activator 
+would probably expose these methods as static methods as follows:
+
+```
+	public static IPluginLog pluginLog() {
+		return getDefault().pluginLogInternal();
+	}
+	public static StatusFactory statusFactory() {
+		return getDefault().statusFactoryInternal();
+	}
+```
+
+== Tracing
+
+Tracing is a mechanism provided by eclipse to help log a subset of messages based on various keys. It is very useful when debugging, because you can 
+select which types of messages you want logged. In runtime-workbench, you can accomplish this via UI, and in a full installation, you can enable this
+by using a .options file in your eclipse installation's root folder. 
+
+For some information on eclipse tracing, see https://wiki.eclipse.org/FAQ_How_do_I_use_the_platform_debug_tracing_facility
+
+In `foundation`, however, we provide a base class `AbstractTrace` which you should extend, as well as a method to assist in wiring up the tracing. 
+
+=== Your Trace Class
+
+At a minimum, your Trace class should look as follows:
+
+```
+public class Trace extends AbstractTrace implements DebugOptionsListener {
```

Comment:
yes, there is no mention of the package so dev would need to go look it up instead of just having that one line included and a copy/paste would just work.